plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -218,16 +218,30 @@ def process_main(self, item=None):
         manifest_type_id = models.Manifest._content_type_id.default
         repo_manifest_ids = repository.get_associated_unit_ids(repo_id, manifest_type_id)
 
-        # check if there is manifest with such id within the queried rpeo
-        manifest = models.Manifest.objects.filter(digest=digest, id__in=repo_manifest_ids)
-        if manifest.count() == 0:
-            raise PulpCodedValidationException(error_code=error_codes.DKR1010,
-                                               digest=digest,
-                                               repo_id=repo_id)
+        # check if there is manifest with such id within the queried repo
+        # since we don't know if the provided digest is of an image manifest or manifest list
+        # we first need to try both
+        #TODO
```

Comment:
i am not sure i want to make it a separate method just for one single place it will be used.
if you are ok with this, i will leave this as it is, but i will leave some doc notes and explanation